### PR TITLE
improved support for SRS signal generators

### PR DESCRIPTION
**Pull request by qku**
## Description

This PR introduces improvements to the microwave hardware file for Stanford Research Systems (SRS) signal generators.

- support for all current SRS signal generators (see [manufacturer website](https://www.thinksrs.com/products/test.html))
- proper matching of the model identifier to supported functions and hardware constraints
- check for error after every write command
- display current frequency during scan
- type annotations and documentation

## Motivation and Context

We are using a non-vector SRS signal generator in our lab. It was not compatible with the current hardware file due to the usage of commands which are defined exclusively for vector models.

## How Has This Been Tested?

Tested on real hardware (SG384) with the ODMR toolchain.

## Types of changes

- [ ] Bug fix
- [X] New feature
- [ ] Breaking change (Causes existing functionality to not work as expected)

## Checklist:

- [X] My code follows the code style of this project.
- [X] I have documented my changes in `/docs/changelog.md`.
- [ ] My change requires additional/updated documentation.
- [X] I have updated the documentation accordingly.
- [ ] I have added/updated the config example for any module docstrings as necessary.
- [X] I have checked that the change does not contain obvious errors
(syntax, indentation, mutable default values, etc.).
- [ ] I have tested my changes using 'Load all modules' on the default dummy configuration.
- [ ] All changed Jupyter notebooks have been stripped of their output cells.

### CODE REVIEW

File: src/qudi/hardware/microwave/mw_source_srssg.py [hunk 1]
```diff
@@ -21,10 +25,8 @@
 """
 
 
-try:
-    import pyvisa as visa
-except ImportError:
-    import visa
+import pyvisa as visa
+from pyvisa.resources import MessageBasedResource
```
Comment (NicolasStaudenmaier) [inline src/qudi/hardware/microwave/mw_source_srssg.py:29]: Is there any particular reason why you changed the imports? I would suggest
```
try:
    import pyvisa as visa
    from pyvisa.resources import MessageBasedResource
except ImportError:
    import visa
    from visa.resources import MessageBasedResource
```

**Review by NicolasStaudenmaier — COMMENTED** [review]
I quickly tested with a SRS384 and it seems to work. I have only tested for possible errors, not for correct functionality. 
Code review looks good to me, except for one comment at the imports. If we make the visa import consistent I approve the PR.

**Comment by qku** [comment]
Thank you for reviewing @NicolasStaudenmaier ! I changed the visa imports as suggested.

**Review by NicolasStaudenmaier — APPROVED** [review]
Looks good. Thanks Kilian for the work on it.

**Pull request merged**